apiserver/facades/agent/agent/register.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2022 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package agent
+
+import (
+	"reflect"
+
+	"github.com/juju/errors"
+
+	"github.com/juju/juju/apiserver/common"
+	"github.com/juju/juju/apiserver/common/cloudspec"
+	apiservererrors "github.com/juju/juju/apiserver/errors"
+	"github.com/juju/juju/apiserver/facade"
+)
+
+// Registry describes the API facades exposed by some API server.
+type Registry interface {
```

Comment:
I know we should declare these close to usage, but I think it's reasonable to have this up in the `apiserver/facade` or `apiserver/facades` package for brevity.